### PR TITLE
starkmatt/c03-docker02

### PR DESCRIPTION
**Pull request by starkmatt**
# Description

Exercise C03 - E02

Student Name: Matt G

This was really fun to do, I added some notes in my troubleshooting txt file

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/03class/exercises/c03-docker02/starkmatt/Dockerfile.v2 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM httpd:2.4.41-alpine
+EXPOSE 80
```
Comment (kikobr82) [inline classes/03class/exercises/c03-docker02/starkmatt/Dockerfile.v2]: I reckon the `/usr/local/apache2/htdocs/` folder is not empty with this Dockerfile...
```
Named exercise02 with a tag v2 with the following pre-requisites
Dockerfile must be named Dockerfile.v2
Final image is based from httpd:2.4.41-alpine
The folder /usr/local/apache2/htdocs/ must be empty
```

File: classes/03class/exercises/c03-docker02/starkmatt/c03-docker02.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
+##Exercise 2###
+
+##Webserver 01 Build/Run
+➜  starkmatt git:(starkmatt/c03-docker02) ✗ docker build -t  exercise02:v1 -f Dockerfile.v1 . 
+➜  starkmatt git:(starkmatt/c03-docker02) ✗ docker run -dp 8081:80 exercise02:v1
+52851b9c34fe4814a8dcdd7202b382209fae8c5005ebe95ad5ef6b5ee54f3acb
+
+##Web Server 02 Build/Run
+➜  starkmatt git:(starkmatt/c03-docker02) ✗ docker build -t  exercise02:v2 -f Dockerfile.v2 . 
+➜  starkmatt git:(starkmatt/c03-docker02) ✗ docker run -dp 8082:80 exercise02:v2
```
Comment (kikobr82) [inline classes/03class/exercises/c03-docker02/starkmatt/c03-docker02.txt]: You're not mapping the required folder with this command.

File: classes/03class/exercises/c03-docker02/starkmatt/Dockerfile.v2 (hunk 1 above)
Comment (starkmatt) [reply classes/03class/exercises/c03-docker02/starkmatt/Dockerfile.v2]: You are right!
Would we run the CMD rmdir -p /usr/local/apache2/htdocs/ in the Dockerbuild file ?

Comment (kikobr82) [reply classes/03class/exercises/c03-docker02/starkmatt/Dockerfile.v2]: you need to run a RUN instruction to execute the removal of the files during the image creation.

File: classes/03class/exercises/c03-docker02/starkmatt/c03-docker02.txt (hunk 2 above)
Comment (starkmatt) [reply classes/03class/exercises/c03-docker02/starkmatt/c03-docker02.txt]: Hey Kiko,
Not to sure what you mean ?

Comment (kikobr82) [reply classes/03class/exercises/c03-docker02/starkmatt/c03-docker02.txt]: in the v2 image there is nothing in the htdocs folder, so you need to map the local folder when running the container:
```
Run a second container using the exercise02:v2 image exposing the container port(80) to your localhost port 8082. Additionaly, map the folder classes/03class/docker/artifacts/c03-docker02/ of this repo into the folder /usr/local/apache2/htdocs/ of the container during it's execution. You should be able to access the webserver throught http://localhost8082
```

**Pull request merged**